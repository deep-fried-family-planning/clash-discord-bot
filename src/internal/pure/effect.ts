import {Order} from 'effect';


export {
    Effect as E,
    Console as CSL,
    Data as D,
    DateTime as DT,
    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,
    Cache as C,
    Logger as Logger,
    LogLevel,
    LogSpan,
    Schema as S,
    Brand as B,
    RateLimiter as RL,
    Metric as Metric,
    Order as ORD,
    Option as O,
    Equal as EQ,

    Cron,

    pipe,
    pipe as p,
    flow,
    flow as f,

} from 'effect';


export {
    gen as g,

} from 'effect/Effect';


export const ORDS = Order.string;
export const ORDSR = Order.reverse(Order.string);
export const ORDN = Order.number;
export const ORDNR = Order.reverse(Order.number);


export * as Ar from 'effect/Array'; // collection array
export * as Kv from 'effect/Record'; // collection key-value
