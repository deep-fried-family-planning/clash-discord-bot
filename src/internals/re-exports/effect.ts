export {
    Effect as E,

    Config as Cfg,
    Redacted as Rdt,

    Console as CSL,

    Data as D,
    DateTime as DT,

    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,

    Cache as C,

    Logger as Logger,
    LogLevel,
    LogSpan,

    Schema as S,

    Brand as B,

    RateLimiter as RL,

    Metric as Metric,

    flow,
    pipe,

} from 'effect';
