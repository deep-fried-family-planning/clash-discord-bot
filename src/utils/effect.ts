export {
    Effect as E,

    Config as Cfg,
    Redacted as Rdt,

    Layer as L,

    Logger as Logger,
    LogLevel,
    LogSpan,

    RateLimiter as RL,

    Metric as Metric,

    flow,
    pipe,

} from 'effect';
