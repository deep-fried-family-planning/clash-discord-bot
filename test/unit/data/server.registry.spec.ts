import {ServerRegistry} from '#src/data/index.ts';
import {mockDb, mockDbLayer} from '#unit/data/mock-db.ts';
import {it} from '@effect/vitest';
import * as E from 'effect/Effect';
import {pipe} from 'effect/Function';

it.effect('when registering a new server', E.fn(function* () {
  mockDb.get.mockReturnValueOnce(E.succeed({Item: undefined}));

  const actual = yield* pipe(
    ServerRegistry.register({
      caller_id   : 'user',
      caller_roles: ['admin'],
      guild_id    : 'guild',
      payload     : {
        admin: 'admin',
      },
    }),
    E.provide(mockDbLayer),
  );

  expect(actual).toMatchSnapshot();
  expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
  expect(mockDb.put.mock.calls[0][0].Item).toMatchSnapshot();
}));

it.effect('when re-registering a server', E.fn(function* () {
  mockDb.get.mockReturnValueOnce(E.succeed({
    Item: {
      _tag             : 'Server',
      admin            : 'admin',
      created          : '1970-01-01T00:00:00.000Z',
      gsi_all_server_id: 's-guild',
      pk               : 's-guild',
      sk               : 'now',
      updated          : '1970-01-01T00:00:00.000Z',
      version          : 0,
    },
  }));

  const actual = yield* pipe(
    ServerRegistry.register({
      caller_id   : 'user',
      caller_roles: ['admin', 'admin2'],
      guild_id    : 'guild',
      payload     : {
        admin: 'admin2',
      },
    }),
    E.provide(mockDbLayer),
  );

  expect(actual).toMatchSnapshot();
  expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
  expect(mockDb.put.mock.calls[0][0].Item).toMatchSnapshot();
}));
