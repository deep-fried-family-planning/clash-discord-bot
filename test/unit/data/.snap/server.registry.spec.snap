// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when re-registering a server 1`] = `
{
  "description": "Success",
}
`;

exports[`when re-registering a server 2`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`when re-registering a server 3`] = `
{
  "_tag": "Server",
  "admin": "admin2",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_all_server_id": "s-guild",
  "pk": "s-guild",
  "sk": "now",
  "updated": "1970-01-01T00:00:00.000Z",
  "version": 0,
}
`;

exports[`when registering a new server 1`] = `
{
  "description": "Success",
}
`;

exports[`when registering a new server 2`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`when registering a new server 3`] = `
{
  "_tag": "Server",
  "admin": "admin",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_all_server_id": "s-guild",
  "pk": "s-guild",
  "sk": "now",
  "updated": "1970-01-01T00:00:00.000Z",
  "version": 0,
}
`;
